File: easy/1108_defanging_an_ip_address/solution.py
class Solution:
    def defangIPaddr(self, address: str) -> str:
        defanged_address = []

        for c in address:
            if c == ".":
                defanged_address.append("[.]")
            else:
                defanged_address.append(c)

        return "".join(defanged_address)

File: easy/1108_defanging_an_ip_address/solution.cpp
#include <string>

class Solution {
public:
  std::string defangIPaddr(std::string address) {
    std::string defangedAddress;

    for (const char &ch : address) {
      if (ch == '.')
        defangedAddress += "[.]";
      else
        defangedAddress.push_back(ch);
    }

    return defangedAddress;
  }
};
